tests/test_fido.py

```diff
@@ -1,13 +1,50 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
+from __future__ import print_function
 
+import io
+import tempfile
 from time import sleep
 
+from fido import fido
 from fido.fido import PerfTimer
 
+# Magic number for fmt/1000.
+MAGIC = b"\x5A\x58\x54\x61\x70\x65\x21\x1A\x01"
 
 def test_perf_timer():
     timer = PerfTimer()
     sleep(3.6)
     duration = timer.duration()
     assert duration > 0
+
+def test_file_identification():
+    """Reference for Fido-based format identification
+        1. Create a byte-stream with a known magic number and serialise to tempfile.
+        2. Call identify_file(...) to identify the file against Fido's known formats.
+    """
+    # Create a temporary file on the host operating system.
+    tmp = tempfile.mkstemp()
+    tmp_file = tmp[1]
+
+    # Write to the file our known magic-number.
+    with open(tmp_file, "wb") as new_file:
+        new_file.write(MAGIC)
+
+    # Create a Fido instance and call identify_file. The identify_file function
+    # will create and manage a file for itself.
+    f = fido.Fido()
+    f.identify_file(tmp_file)
```

Comment:
Nice tests, Carl. I think this one could be a good use case for the [`tmp_path` fixture](https://docs.pytest.org/en/latest/tmpdir.html#the-tmp-path-fixture) provided by pytest. It gives us a `pathlib.Path` object which has a nice API. pytest also takes care of removing old temporary directories, always [keeping the last three](https://docs.pytest.org/en/latest/tmpdir.html#the-default-base-temporary-directory).

```suggestion
def test_file_identification(tmp_path):
    """Reference for Fido-based format identification
        1. Create a byte-stream with a known magic number and serialise to tempfile.
        2. Call identify_file(...) to identify the file against Fido's known formats.
    """
    # Create a temporary file on the host operating system.
    tmp_file = tmp_path / "file"

    # Write to the file our known magic-number.
    tmp_file.write_bytes(MAGIC)

    # Create a Fido instance and call identify_file. The identify_file function
    # will create and manage a file for itself.
    f = fido.Fido()
    f.identify_file(str(tmp_file))
```

Reply:
So the original ideas here came from @ross-spencer (that's intended to give credit not share blame for temp file management). I agree that these could be better for test purposes. The PR will remain draft for now as I'm going to test a couple of ideas around the API.